### PR TITLE
Features/52706 update problem details rule

### PR DESCRIPTION
**Pull request by christoment**
As per task in TimePro board [Product Backlog Item 52680](https://dev.azure.com/ssw2/SSW.TimeProDotNet/_workitems/edit/52680): ⚖️ SSW Rule | Rule on error handling on frontend

this PR updates the rule to detailed error message with ProblemDetails in modern .NET

- Added ProblemDetails specification
- Added options on how to return ProblemDetails in .NET

### CODE REVIEW

File: rules/do-you-return-detailed-error-messages/rule.md [hunk 1]
```diff
@@ -17,23 +19,103 @@ Good error design is as important to the sucess of an API as the API design itse
...
-* **400 Bad Request** - The consuming application did something wrong.
-* **500 Internal Server Error** - The API Application did something wrong.
 
-### ...And then include the problem details
-RFC 7807 - Problem Details for HTTP APIs (ietf.org) details the specification for returning errors from your API. The HTTP Status Codes are an excellent start - they immediately tell you *where*the problem is, but they don't tell you *what*the problem is.
+### Use ProblemDetails Format
+[RFC 7807 - Problem Details for HTTP APIs](https://www.rfc-editor.org/rfc/rfc7807.html) details the specification for returning errors from your API. 
 
-ASP.Net Core has built in support for the problem details specification. You can see more at the official documentation. Handle errors in ASP.NET Core web APIs | Microsoft Docs
+Problem Details defines a standardised way for an HTTP APIs to communicate errors to clients. It introduces a simple and consistent format for describing errors, providing developers with a clear and uniform way to understand and handle errors in HTTP APIs.
```
Comment (matt-goldman) [inline rules/do-you-return-detailed-error-messages/rule.md:39]: > an HTTP APIs
Should be an HTTP API or HTTP APIs

But also it's not just APIs.

File: rules/do-you-return-detailed-error-messages/rule.md [hunk 2]
```diff
@@ -17,23 +19,103 @@ Good error design is as important to the sucess of an API as the API design itse
...
+
+In the above example:
+- `type` specifies a URI that uniquely identifies the type of the problem.
+- `title` provides a short, human-readable summary of the problem.
+- `status` indicates the HTTP status code for the response.
+- `detail` gives a human-readable explanation specific to the occurrence of the problem.
+- `instance` provides a URI reference that identifies the specific occurrence of the problem.
+- `allowedCharacters` is an example property specificly added to the problem.
+
+Using the above structured message, APIs can now reliably communicate with clients to provide a better error handling.
```
Comment (matt-goldman) [inline rules/do-you-return-detailed-error-messages/rule.md:61]: I would change this to something like:
Using the above structured message format, APIs can now reliably communicate problems to clients to enable better error handling.

File: rules/do-you-return-detailed-error-messages/rule.md [hunk 3]
```diff
@@ -17,23 +19,103 @@ Good error design is as important to the sucess of an API as the API design itse
...
+app.UseExceptionHandler();
+```
+
+Using this option, the API will generate a problem details response for all HTTP client and server error responses that *don't have a body content yet*.
+
+You can also customise the `ProblemDetailsService` behaviour - read more about it in the following link [Handle errors in ASP.NET Core | Customise Problem Details](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-8.0#customize-problem-details).
+
+
+⚠️ **Important**
+When there is an exception, this approach will responds in ProblemDetails format **only in Production mode**!
```
Comment (matt-goldman) [inline rules/do-you-return-detailed-error-messages/rule.md:88]: I would change this to:
The built in .NET ProblemDetails service will **only** produce ProblemDetails responses for exceptions in release configuration.

File: rules/do-you-return-detailed-error-messages/rule.md [hunk 4]
```diff
@@ -17,23 +19,103 @@ Good error design is as important to the sucess of an API as the API design itse
...
+Option 2 provides more flexibility to control this behaviour.
+
+
+### Option 2 - Customise Exception Handler Middleware (Recommended)
+
+This option provides more flexibility in controlling how the ProblemDetails.
```
Comment (matt-goldman) [inline rules/do-you-return-detailed-error-messages/rule.md:94]: > This option provides more flexibility in controlling how the ProblemDetails.

Sentence ends abruptly

**Review by matt-goldman — CHANGES_REQUESTED** [review]
This is a great improvement to the rule! Some minor tweaks suggested.

File: rules/do-you-return-detailed-error-messages/rule.md (hunk 1 above)
Comment (christoment) [reply rules/do-you-return-detailed-error-messages/rule.md]: Thanks, nice spot. I've rechecked the whole content for grammatical error so hopefully no more silly error this time! 😄

File: rules/do-you-return-detailed-error-messages/rule.md (hunk 2 above)
Comment (christoment) [reply rules/do-you-return-detailed-error-messages/rule.md]: Done!

File: rules/do-you-return-detailed-error-messages/rule.md (hunk 3 above)
Comment (christoment) [reply rules/do-you-return-detailed-error-messages/rule.md]: Done - updated for more clarity

File: rules/do-you-return-detailed-error-messages/rule.md (hunk 4 above)
Comment (christoment) [reply rules/do-you-return-detailed-error-messages/rule.md]: I've added more explanation to this 😄

**Review by matt-goldman — COMMENTED** [review]
LGTM 🚀

**Review by matt-goldman — APPROVED** [review]

**Pull request merged**